src/dialects/db2/data-types.ts

```diff
@@ -0,0 +1,380 @@
+  }
+
+  static parse(value: RawTypeOf<BaseTypes.DATE>) {
+    if (typeof value !== 'string') {
+      // @ts-expect-error what type is value?
+      value = value.string();
+    }
+
+    if (value === null) {
+      return value;
+    }
+
+    // @ts-expect-error what type is value?
+    value = new Date(momentTz.utc(value));
+
+    return value;
+  }
+}
+
+export class DATEONLY extends BaseTypes.DATEONLY {
+  static parse(value: RawTypeOf<BaseTypes.DATEONLY>) {
+    return momentTz(value).format('YYYY-MM-DD');
+  }
+}
+
+export class INTEGER extends BaseTypes.INTEGER {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.INTEGER>) {
+    super(...args);
+    removeUnsupportedIntegerOptions(this);
+  }
+}
+
+export class TINYINT extends BaseTypes.TINYINT {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.TINYINT>) {
+    super(...args);
+    removeUnsupportedIntegerOptions(this);
+  }
+}
+
+export class SMALLINT extends BaseTypes.SMALLINT {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.SMALLINT>) {
+    super(...args);
+    removeUnsupportedIntegerOptions(this);
+  }
+}
+
+export class BIGINT extends BaseTypes.BIGINT {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.BIGINT>) {
+    super(...args);
+    removeUnsupportedIntegerOptions(this);
+  }
+}
+
+export class REAL extends BaseTypes.REAL {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.REAL>) {
+    super(...args);
+    // Db2 does not support any options for real
+    if (
+      this._length
+      || (this.options.length != null && this.options.length > 0)
+      || this._unsigned
+      || this._zerofill
+    ) {
+      warn(
+        'Db2 does not support REAL with options. Plain `REAL` will be used instead.',
+      );
+      this._length = undefined;
+      this.options.length = undefined;
+      this._unsigned = undefined;
+      this._zerofill = undefined;
+    }
+  }
+}
+
+export class FLOAT extends BaseTypes.FLOAT {
+  constructor(...args: ConstructorParameters<typeof BaseTypes.FLOAT>) {
+    super(...args);
+    // Db2 does only support lengths as option.
+    // Values between 1-24 result in 7 digits precision (4 bytes storage size)
+    // Values between 25-53 result in 15 digits precision (8 bytes size)
+    // If decimals are provided remove these and print a warning
+    if (this._decimals) {
+      warn(
+        'Db2 does not support Float with decimals. Plain `FLOAT` will be used instead.',
+      );
+      this._length = undefined;
+      this.options.length = undefined;
+    }
+
+    if (this._unsigned) {
+      warn('Db2 does not support Float unsigned. `UNSIGNED` was removed.');
+      this._unsigned = undefined;
+    }
+
+    if (this._zerofill) {
+      warn('Db2 does not support Float zerofill. `ZEROFILL` was removed.');
+      this._zerofill = undefined;
+    }
+  }
+}
+
+export class ENUM<Member extends string> extends BaseTypes.ENUM<Member> {
+  toSql() {
+    return 'VARCHAR(255)';
+  }
+}
```

Comment:
Should we drop support for this entirely?  It seems misleading to support it as a string.

Reply:
I agree, its constructor should throw & say to use DataTypes.STRING + the `isIn` column validator instead

Reply:
My only concern with this is that it'll be pretty annoying to change w/ an upgrade.

Thoughts on leaving this for now and then in a different change add an  `ENUM_VIRTUAL` datatype + deprecate fake enum usage?